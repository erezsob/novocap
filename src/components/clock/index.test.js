import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Clock from './';

jest.mock('./../../util/time', () => ({
  time: jest.fn().mockReturnValue('12:34:32'),
  date: jest.fn().mockReturnValue('Wed, 12/05/2030')
}));

describe('<Clock />', () => {
  test('renders with the right css classes', () => {
    const wrapper = shallow(<Clock />);
    expect(wrapper.find('.container')).toHaveLength(1);
    expect(wrapper.find('.time')).toHaveLength(1);
    expect(wrapper.find('.date')).toHaveLength(1);
  });

  test('renders properly', () => {
    const tree = renderer.create(<Clock />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
