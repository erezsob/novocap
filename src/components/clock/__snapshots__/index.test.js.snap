// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> renders properly 1`] = `
<div
  className="container"
>
  <div
    className="time"
  >
    12:34:32
  </div>
  <div
    className="date"
  >
    Wed, 12/05/2030
  </div>
</div>
`;
